zipline/assets/assets.py

```diff
@@ -338,6 +446,8 @@ def spawn_asset(self, identifier, **kwargs):
         self.identifier_cache[identifier] = asset
         if asset.symbol is not '':
             self.sym_cache.setdefault(asset.symbol, []).append(asset)
+        if isinstance(asset, Future) and asset.root_symbol is not '':
+            self.future_cache.setdefault(asset.root_symbol, []).append(asset)
```

Comment:
Ah, @jfkirk I misunderstood your suggestion.  I thought you were just saying we should change the branch from `if isinstance(asset, Future)` to `if asset_type == 'future'`.  I agree that we should move this branch up so that there's only one place that we have "futures-only" logic.